packages/drivers/odsp-driver/src/odspCache.ts

```diff
@@ -21,79 +21,46 @@ export interface IPersistedFileCache {
     removeEntries(): Promise<void>;
 }
 
-/**
- * Handles garbage collection of expiring cache entries.
- * Not exported.
- * (Based off of the same class in promiseCache.ts, could be consolidated)
- */
-class GarbageCollector<TKey> {
-    private readonly gcTimeouts = new Map<TKey, ReturnType<typeof setTimeout>>();
-
-    constructor(
-        private readonly cleanup: (key: TKey) => void,
-    ) { }
-
-    /**
-     * Schedule GC for the given key, as applicable
-     */
-    public schedule(key: TKey, durationMs: number) {
-        this.gcTimeouts.set(
-            key,
-            setTimeout(
-                () => { this.cleanup(key); this.cancel(key); },
-                durationMs,
-            ),
-        );
-    }
-
-    /**
-     * Cancel any pending GC for the given key
-     */
-    public cancel(key: TKey) {
-        const timeout = this.gcTimeouts.get(key);
-        if (timeout !== undefined) {
-            clearTimeout(timeout);
-            this.gcTimeouts.delete(key);
-        }
-    }
-}
-
 /**
  * Default local-only implementation of IPersistedCache,
  * used if no persisted cache is provided by the host
  */
 export class LocalPersistentCache implements IPersistedCache {
-    private readonly cache = new Map<string, any>();
-    private readonly gc = new GarbageCollector<string>((key) => this.cache.delete(key));
+    private readonly pc: PromiseCache<string, any>;
 
-    public constructor(private readonly snapshotExpiryPolicy = 30 * 1000) {}
+    public constructor(private readonly snapshotExpiryPolicy = 30 * 1000) {
+        this.pc = new PromiseCache<string, any>({ expiry: {policy:"sliding", durationMs: this.snapshotExpiryPolicy} });
```

Comment:
there's a minor issue with the spaces and formatting of this json. VSCode should take care of that for you

Reply:
I might be missing some extension as it did not fix it - will double check.

Reply:
@andre4i - I also don't have an extension for this, what's your secret?

Reply:
Not sure it's because of an extension. The only extra thing I have installed is `eslint` but it's probably configured poorly as it doesn't match the project's lint rules. This should work out of the box for javascript/ts/json: https://code.visualstudio.com/docs/editor/codebasics#_formatting